controllers/user.js

```diff
@@ -226,45 +225,73 @@ async function verify(req, res) {
 }
 
 
+// legacy /user/unsubscribe controller for pre-FxA unsubscribe links
 async function getUnsubscribe(req, res) {
   if (!req.query.token) {
     throw new FluentError("user-unsubscribe-token-error");
   }
 
   const subscriber = await DB.getSubscriberByToken(req.query.token);
+  // Token is for a primary email address,
+  // redirect to preferences to remove Firefox Monitor
+  if (subscriber) {
+    return res.redirect("/user/preferences");
+  }
 
-  //throws error if user backs into and refreshes unsubscribe page
-  if (!subscriber) {
+  const emailAddress = await DB.getEmailByToken(req.query.token);
+  if (!subscriber && !emailAddress) {
```

Comment:
Not sure about this logic.
If `subscriber` is truthy on L237-239 above, we `return` and exit this function, which means that the `!subscriber` check is probably always true here. Maybe kinda?

Reply:
Good catch.